pkg/scheduler/scheduler.go

```diff
@@ -915,7 +915,9 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 			diagnosis.NodeToStatusMap[n.Node().Name] = s
 		}
 		// Status satisfying IsUnschedulable() gets injected into diagnosis.UnschedulablePlugins.
-		diagnosis.UnschedulablePlugins.Insert(s.FailedPlugin())
+		if s.FailedPlugin() != "" {
+			diagnosis.UnschedulablePlugins.Insert(s.FailedPlugin())
```

Comment:
Since there is no failed plugin, could this affect re-scheduling?

Reply:
it will place it back in the queue on all events.